exercise_4/src/main.cpp

```diff
@@ -0,0 +1,125 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <map>
+#include <sstream>
+#include <vector>
+#include <memory>
+
+#include "IRecord.h"
+#include "Course.h"
+#include "Exam.h"
+#include "Student.h"
+#include "Teacher.h"
+#include "CommandHandler.h"
+
+using namespace std;
```

Comment:
Don't use `using namespace ...`, especially the `std` one.
https://stackoverflow.com/questions/1452721/why-is-using-namespace-std-considered-bad-practice

Reply:
Removed.